nats/example_test.go

```diff
@@ -21,26 +20,37 @@ func ExamplePublisher() {
 }
 
 func ExampleSubscriber() {
-	ctx := context.TODO()
-	sub, err := bps.NewSubscriber(ctx, "nats://localhost:4222/?client_id=my_client&start_at=first")
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	subscriber, err := bps.NewSubscriber(ctx, "nats://localhost:4222/?client_id=my_client&start_at=first")
 	if err != nil {
 		panic(err.Error())
 	}
-	defer sub.Close()
-
-	// can be just context.WithCancel - `cancel` can be used to stop consuming:
-	ctx, cancel := context.WithTimeout(ctx, time.Second)
-	defer cancel()
+	defer subscriber.Close()
 
-	// will block till context is cancelled:
-	err = sub.Topic("topic").Subscribe(
-		ctx,
+	var subscription bps.Subscription
+	subscription, err = subscriber.Topic("topic").Subscribe(
```

Comment:
just `subscription, err := `. don't close the subscription inside the handler, let's not do that or suggest that it's a good idea. I'd rather do the (n)acking later on `bps.SubMessage` instead